src/Components/ServiceScene/LogsJsonScene.tsx

```diff
@@ -399,8 +400,28 @@ export class LogsJsonScene extends SceneObjectBase<LogsJsonSceneState> {
     );
   };
 
-  public renderCopyToClipboardButton(keyPath: KeyPath) {
-    return <CopyToClipboardButton onClick={() => copyLogLine(keyPath, sceneGraph.getData(this))} />;
+  private getLinkToLog(keyPath: KeyPath) {
+    const timeRange = sceneGraph.getTimeRange(this).state.value;
+    const dataFrame = this.state.rawFrame;
+    const idField: Field<string> | undefined = dataFrame?.fields.find((f) => isLogsIdField(f.name));
+    const logLineIndex = keyPath[0];
+    if (!isNumber(logLineIndex)) {
+      const error = Error('Invalid line index');
+      logger.error(error, { msg: 'Error getting log line index' });
+      throw error;
+    }
+    const logId = idField?.values[logLineIndex];
+    const logLineLink = generateLogShortlink('selectedLine', { id: logId, row: logLineIndex }, timeRange);
```

Comment:
A reminder that this matches the behavior that the table introduced but not what is used by the visualization, supported in Dashboards and Explore, using `panelState`

Reply:
Still waking up here, is that a good thing or no? 😄

Reply:
No, I think all the viz should use panelState, no only to be consistent between them but also DD/Dashboards/Explore. I thought we had an issue for that, but I couldn't find it with a quick search.